### PR TITLE
cleans up and brings up to date documentation on SEO

### PR DESCRIPTION
**Pull request by jasonfb**

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://jasonfb-master.spree-spree-9ed724b77adec04c49c2.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://jasonfb-master.spree-spree-9ed724b77adec04c49c2.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by jasonfb** [comment]
this is wrong -- do not merge

**Comment by jasonfb** [comment]
ok @JDutil ready now. All this information is accurate to the best of my knowledge. In general I'm trying to stick more information about specific Spree versions and when things get introduced to make these guides more useful to people as they upgrade. Please let me know if you have any feedback!

**Comment by JDutil** [comment]
Thanks specific version info should definitely help.  We've been updating our infrastructure hosting our sites, and ideally we should be deploying out guides for each stable branch like 2-4-stable-guides.spreecommerce.com/ soon. They used to link between version, but that was lost along the way over the years for some reason.

File: guides/content/developer/advanced/seo.markdown [hunk 1]
```diff
@@ -106,16 +82,33 @@ Webmaster Tools and Quantcast. Social media optimization tools such as
...
+
+Although several common search engine optimization issues have been
+addressed, we are always looking for the new best practices in SEO.
+Contributions to address issues will be very welcome. Visit the
+[contributing to spree section](contributing.html) to learn
+more about contributing.
+
+Here are some of the specific planned ideas we have for the future of Spree:
+
+- Make the `alt` field from spree_assets output as the alt attribute in the image tag
```
Comment (JDutil) [inline guides/content/developer/advanced/seo.markdown]: wow we don't already do this?

**Comment by jasonfb** [comment]
re: "for each stable branch"

Personally I would rather just edit one guide and mark the version differences inline with the text (I know this may be different from how I felt in the past). Also that might a little easier for the reader instead of having to check different guides. 

Maybe as Spree grows we could do a compromise, say, one guide to cover the whole 2.x line?

re: "Alt text"
It doesn't work in my 2.2, admittedly I didn't check higher versions.

**Pull request closed without merging**